src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -338,7 +328,10 @@ class LoopOpRewriteMaxTripCountPattern : public OpRewritePattern<ONNXLoopOp> {
     int64_t lowerBound = getOneIntergerConstant(startValue);
     int64_t upperBound = getOneIntergerConstant(ubValue);
     int64_t step = getOneIntergerConstant(stepValue);
-    int64_t derivedTripCount = (int64_t)((upperBound - lowerBound) / step);
+    if (step == 0)
+      return std::make_pair(false, -1);
+    int64_t derivedTripCount = (upperBound - lowerBound) / step +
+                               ((upperBound - lowerBound) % step != 0);
```

Comment:
I am afraid this algorithm does not work for negative divisor. Best using what I used in IndexExpr.cpp, namely
```
    int64_t rval = ceil((1.0 * aa.getLiteral()) / (1.0 * bb.getLiteral()));
```

Reply:
Done. I changed to use ceil. Thanks!